pkg/valueset/config.go

```diff
@@ -0,0 +1,118 @@
+package valueset
+
+import (
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/cloudprivacylabs/lsa/layers/cmd/cmdutil"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
+	ValuesetDBs []ValuesetDB
+}
+
+/*
+	Databases []struct {
+		postgresql.Database `json:"database" yaml:"database"`
+	} `json:"databases" yaml:"databases"`
+*/
+
+type ValuesetDB interface {
+	ValueSetLookup(string, map[string]string) (map[string]string, error)
+	GetTableIds() []string
+}
+
+var DatabaseFactory = map[string]func(interface{}) (ValuesetDB, error){
```

Comment:
rename to valuesetFactory
Initialize in pgx.init()
The factory method should unmarshal the argument
marshal to json, and unmarshal it
You need to pass the env to the factory, so:

func(interface{},map[string]string) (ValuesetDB, error)